### PR TITLE
Refactored src/components/AddOn/support/components/Action/Action.test.tsx from Jest to Vitest

### LINKED ISSUE #2474: Refactor:  `src/components/AddOn/support/components/Action/Action.test.tsx`  from Jest to Vitest
**Is your feature request related to a problem? Please describe.**

This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor this file's test cases to align with Vitest's syntax and features.

**Acceptance Criteria**

1. Replace Jest-specific functions and mocks with Vitest equivalents
2. Ensure all tests in `src/components/AddOn/support/components/Action/Action.test.tsx`  pass after migration using `npm run test:vitest`
3. Maintain the test coverage for the file as 100% after migration 
4. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional Context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation: 
  https://vitest.dev/

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by adithyanotfound**
**What kind of change does this PR introduce?**

Refactoring

**Issue Number:**

Fixes #2474 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

Screenshot 2024-11-27 at 11 58 08 AM

Screenshot 2024-11-27 at 3 36 44 PM

**Summary**

1. Refactored tests from jest to vitest
2. Added vite.setup.ts to include `@testing-library/jest-dom` to support method `toBeInTheDocument()`

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **Tests**
	- Introduced a new test file for the `Action` component to enhance unit testing.
	- Updated testing configuration to include a setup file for custom logic.
	- Integrated custom matchers from `@testing-library/jest-dom` to improve assertion capabilities in tests.

### CODE REVIEW

File: src/components/AddOn/support/components/Action/Action.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { describe, test, expect } from 'vitest'; // Import from Vitest
+
+import { store } from 'state/store';
+import Action from './Action';
+
```
[the remark this answers is not in the document]
Comment (adithyanotfound) [reply src/components/AddOn/support/components/Action/Action.spec.tsx:8]: It is present in vite.setup.ts

File: src/components/AddOn/support/components/Action/Action.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    const { getByText } = render(
+      <Provider store={store}>
+        <Action {...props} />
+      </Provider>,
+    );
+
+    expect(getByText(props.label)).toBeInTheDocument();
+    expect(getByText(props.children)).toBeInTheDocument();
+  });
+});
```
[the remark this answers is not in the document]
Comment (adithyanotfound) [reply src/components/AddOn/support/components/Action/Action.spec.tsx:25]: Outside scope of PR

**Comment by varshith257** [comment]
@adithyanotfound Yes, add them and you can take reference of the PR linked in the issue

File: vitest.config.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export default defineConfig({
     include: ['src/**/*.spec.{js,jsx,ts,tsx}'],
     globals: true,
     environment: 'jsdom',
+    setupFiles: 'vitest.setup.ts',
```
Comment (varshith257) [inline vitest.config.ts:18]: Does we need this?

Comment (adithyanotfound) [reply vitest.config.ts:18]: Yes. We currently need this for `toBeInTheDocument()` which is used frequently in multiple test files. In future we may need this to support methods from other external libraries.

**Review by varshith257 — APPROVED** [review]

**Comment by varshith257** [comment]
@palisadoes We need to open issues for other test file to migrate

**Comment by palisadoes** [comment]
@varshith257 Can you do that?

**Comment by varshith257** [comment]
@palisadoes No! My University exams are ongoing now and I delegated the same to @prayanshch(well he also has exams).

I will redirect this to any other contributor

**Comment by palisadoes** [comment]
@varshith257 

1. There are 177 `test.tsx` files in the repo. That's too many issues to create. 
2. How about creating issues just for screens and include all the components each uses? That would be faster as it would require only 60 issues.

**Comment by varshith257** [comment]
Yes, that makes sense

**Pull request merged**